cx/slack/test/server_test.js

```diff
@@ -0,0 +1,65 @@
+const detectIntentToSlackMessage = require('../server.js').detectIntentToSlackMessage;
+const slackToDetectIntent = require('../server.js').slackToDetectIntent;
+const assert = require('assert');
+
+describe('slackToDetectIntent()', () => {
+    chatId = 12435;
+
+    const slackRequest = {
+        type: 'message',
+        user: chatId,
+        text: 'Test Text'
+    };
+
+    const dialogflowRequest = {
+        session: 'sessionPath',
+        queryInput: {
+            text: {
+                text: 'Test Text',
+            },
+            languageCode: 'en',
+        },
+    };
+
+    it('should convert Slack request to detectIntent request', async function () {
+        assert.deepStrictEqual(slackToDetectIntent(
+            slackRequest,'sessionPath'), dialogflowRequest)
+    });
+});
+
+describe('detectIntentToSlackMessage()', () => {
+    chatId = 12435;
+    channel_id = 54321
+
+    const slackRequestIm = {
+        channel: chatId,
+        text: 'Test Response Text\n'
+    };
+    
+    const slackRequestChannel = {
+        channel: channel_id,
+        text: 'Test Response Text\n'
+    };
+
+    const dialogflowResponse = {
+        queryResult: {
+            text: 'Test Text',
+            languageCode: 'en',
+            responseMessages: [{
+                text: {
+                    text: 'Test Response Text'
+                }
+            }]
+        }
+    };
+
+    it('should convert detectIntent response to a Slack text message request.', async function () {
```

Comment:
These two test cases don't cover the event type logic, since its behavior actually makes no different for different event types.

It might be better to have the tests covering the event types.
E.g.,
Input an event with `event.channel_type = 'im'` (rather than just an ID), and verify the behavior is as expected.

Reply:
With the restructure of the server.js file and the integration is now able to use app_mention so these tests are no longer needed.